src/app/[locale]/viewer/_components/submodel-elements/generic-elements/ReferenceComponent.tsx

```diff
@@ -0,0 +1,104 @@
+import { Typography, Box } from '@mui/material';
+import ArrowForwardIcon from '@mui/icons-material/ArrowForward';
+import { Reference } from '@aas-core-works/aas-core3.0-typescript/types';
+import { useTranslations } from 'next-intl';
+
+type ReferenceProps = {
+    readonly reference: Reference;
+    readonly showAllKeys?: boolean; // Optional prop to control key display
+};
+
+export function ReferenceComponent(props: ReferenceProps) {
+    const { showAllKeys = true } = props;
```

Comment:
You use the deconstruction pattern here first, but later access the reference through `props.reference`. I would use one or the other style, but not mix them up
```suggestion
export function ReferenceComponent({ reference, showAllKeys = true }: ReferenceProps) {
```